src/main/java/org/apache/sling/jcr/resource/internal/helper/jcr/JcrResourceProvider.java

```diff
@@ -103,6 +106,15 @@
         IGNORED_PROPERTIES.add("jcr:createdBy");
     }
 
+    @ObjectClassDefinition(name = "Apache Sling JCR Resource Provider", description = "Provides Sling resources based on the Java Content Repository")
+    public @interface Config {
+
+        @AttributeDefinition(name = "Enable Query Limit", description = "If set to true, the JcrResourceProvider will set a default limit for all other queries using the findResources / queryResources methods")
+        boolean enable_query_limit() default false;
+
+        @AttributeDefinition(name = "Query Limit", description = "The default query limit for queries using the findResources / queryResources methods")
```

Comment:
```suggestion
        @AttributeDefinition(name = "Query Limit", description = "The default query limit for queries using the findResources / queryResources methods. It will only take effect if the query limit feature is enabled.")
```